tests/unit/metadata/test_single_table.py

```diff
@@ -1033,16 +1112,100 @@ def test_set_sequence_index_validation(self):
         with pytest.raises(ValueError, match=err_msg):
             instance.set_sequence_index(('column1', 'column2'))
 
+    def test_set_sequence_index_validation_columns(self):
+        """Test that ``set_sequence_index`` crashes for invalid arguments.
+
+        Setup:
+            - A ``SingleTableMetadata`` instance with ``_columns`` set.
+
+        Input:
+            - A string not present in ``_columns``.
+
+        Side Effect:
+            - A ``ValueError`` should be raised.
+        """
+        # Setup
+        instance = SingleTableMetadata()
+        instance._columns = {'a', 'd'}
+
+        err_msg = (
+            "Unknown sequence index value {'column'}."
+            ' Keys should be columns that exist in the table.'
+        )
+        # Run / Assert
+        with pytest.raises(ValueError, match=err_msg):
+            instance.set_sequence_index('column')
+
     def test_set_sequence_index(self):
-        """Test that ``set_sequence_index`` sets the ``_metadata['sequence_index']`` value."""
+        """Test that ``set_sequence_index`` sets the ``_sequence_index`` value."""
         # Setup
         instance = SingleTableMetadata()
+        instance._columns = {'column'}
 
         # Run
         instance.set_sequence_index('column')
 
         # Assert
-        assert instance._metadata['sequence_index'] == 'column'
+        assert instance._sequence_index == 'column'
+
+    def test_validate_sequence_index_not_in_sequence_key(self):
+        """Test the ``_validate_sequence_index_not_in_sequence_key`` method."""
+        # Setup
+        instance = SingleTableMetadata()
+        instance._sequence_key = ('abc', 'def')
+        instance._sequence_index = 'abc'
+
+        err_msg = (
+            "'sequence_index' and 'sequence_key' have the same value {'abc'}."
+            ' These columns must be different.'
+        )
+        # Run / Assert
+        with pytest.raises(ValueError, match=err_msg):
+            instance._validate_sequence_index_not_in_sequence_key()
+
+    def test_validate(self):
```

Comment:
I think we should have some of the mocks raise errors and assure that the error messages are correctly concatenated

Reply:
That's why I created the integration test `test_validate_errors`, which prints the big error message with basically all the errors, and checks that everything is formatted properly. Any problem with the errors will more easily be caught there than through any mocks I feel.

Let me know if you still think this change necessary.

Reply:
It doesn't need to be as in depth, but for test coverage purposes I think we need it. The following line only gets entered if there are errors
https://github.com/sdv-dev/SDV/pull/930/files#diff-c15919d0f7ddead675009beb9282fab906be3b8b3232e74ec347332563dcf4faR407